### PR TITLE
Add delegations to admin

### PR DESCRIPTION
**Pull request by sauloperez**
First steps for https://github.com/coopdevs/decidim-coopcat/issues/3

- [x] Create delegations model
- [x] List delegations in a table
- [x] Enable delegations deletion
- [x] Replace `link_to` with `icon_link_to`
- [x] Add menu entry in Participants sidebar
- [x] Permissions
- [x] Scope delegations under organization
- [x] DB indexes

Peek 2020-08-07 09-04

### CODE REVIEW

File: app/views/decidim/action_delegator/admin/delegations/index.html.erb [hunk 1]
```diff
@@ -1 +1,31 @@
...
+        </thead>
+        <tbody>
+          <% delegations.each do |delegation| %>
+            <tr data-delegation-id="<%= delegation.id %>">
+              <td><%= delegation.granter.name %></td>
+              <td><%= delegation.grantee.name %></td>
+              <td><%= l delegation.created_at, format: :short %></td>
+
+              <td class="table-list__actions">
+                <%= link_to t(".delete"), Decidim::ActionDelegator::AdminEngine.routes.url_helpers.delegation_path(delegation), method: :delete %>
```
Comment (sauloperez) [inline app/views/decidim/action_delegator/admin/delegations/index.html.erb:23]: I guess we need to include the url_helpers in the engine file or somewhere else so we can use `delegation_path(delegation)` instead. Any idea @microstudi ?

File: db/migrate/20200729194540_create_decidim_action_delegator_delegations.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateDecidimActionDelegatorDelegations < ActiveRecord::Migration[5.2]
+  def change
+    create_table :decidim_action_delegator_delegations do |t|
+      t.belongs_to :granter
+      t.belongs_to :grantee
```
Comment (sauloperez) [inline db/migrate/20200729194540_create_decidim_action_delegator_delegations.rb:7]: these are the first names that came to mind but suggestions are welcome

File: app/controllers/decidim/action_delegator/admin/delegations_controller.rb [hunk 3]
```diff
@@ -6,10 +6,31 @@ module Admin
       class DelegationsController < ActionDelegator::Admin::ApplicationController
         include NeedsPermission
 
+        # FIXME: the helper method is added but it's not accessible from the template
+        helper Decidim::Admin::IconLinkHelper
```
Comment (sauloperez) [inline app/controllers/decidim/action_delegator/admin/delegations_controller.rb:10]: has this happened to you before @microstudi ?

File: app/views/decidim/action_delegator/admin/delegations/index.html.erb (hunk 1 above)
Comment (microstudi) [reply app/views/decidim/action_delegator/admin/delegations/index.html.erb:23]: that why we create a global module, it create a global object to hold those routes. in this case `decidim_admin_action_delegator.delegation_path(delegation)`

File: app/controllers/decidim/action_delegator/admin/application_controller.rb [hunk 4]
```diff
@@ -6,6 +6,9 @@ module Admin
       # This controller is the abstract class from which all other controllers of
       # this engine inherit.
       class ApplicationController < Decidim::Admin::ApplicationController
+        def permission_class_chain
+          [::Decidim::ActionDelegator::Permissions] + super
+        end
```
Comment (verarojman) [inline app/controllers/decidim/action_delegator/admin/application_controller.rb:11]: Perhaps this file could be refactored to follow the usual style for `permission_class_chain`, like in these files:

- https://github.com/decidim/decidim/tree/develop/decidim-initiatives/app/controllers/decidim/initiatives/admin/application_controller.rb
- https://github.com/decidim/decidim/blob/develop/decidim-participatory_processes/app/controllers/decidim/participatory_processes/application_controller.rb

```suggestion
        register_permissions(::Decidim::ActionDelegator::Admin::ApplicationController,
                             ::Decidim::ActionDelegator::Permissions,
                             ::Decidim::Admin::Permissions)
        
        def permission_class_chain
          ::Decidim.permissions_registry.chain_for(::Decidim::ActionDelegator::Admin::ApplicationController)
        end
```

File: spec/controllers/decidim/action_delegator/admin/delegations_controller_spec.rb [hunk 5]
```diff
@@ -18,7 +18,28 @@ module Admin
           sign_in user
         end
 
+        describe "#index" do
+          it "authorizes the action" do
+            expect(controller).to receive(:allowed_to?).with(:index, :delegation, {})
```
Comment (verarojman) [inline spec/controllers/decidim/action_delegator/admin/delegations_controller_spec.rb:23]: I would rewrite these expectations to the more comprehensive:

```suggestion
            expect(controller.allowed_to? :index, :delegation).to eq true
```

File: spec/controllers/decidim/action_delegator/admin/delegations_controller_spec.rb [hunk 6]
```diff
@@ -18,7 +18,28 @@ module Admin
...
+            get :index
+
+            expect(response).to render_template(:index)
+            expect(response).to have_http_status(:ok)
+          end
+        end
+
         describe "#destroy" do
+          it "authorizes the action" do
+            expect(controller).to receive(:allowed_to?).with(:destroy, :delegation, {})
```
Comment (verarojman) [inline spec/controllers/decidim/action_delegator/admin/delegations_controller_spec.rb:51]: *(following from previous suggestion)*

...and 

```suggestion
            expect(controller.allowed_to? :destroy, :delegation).to eq true
```

**Pull request merged**